### PR TITLE
Idiom fixes

### PR DESCRIPTION
**Pull request by ice1000**
Make sure idiom works

### CODE REVIEW

**Comment by ice1000** [comment]
Fix a part of #561 

bors r+

File: base/src/main/java/org/aya/concrete/desugar/Desugarer.java [hunk 1]
```diff
@@ -34,14 +34,14 @@ private int levelVar(@NotNull Expr expr) throws DesugarInterruption {
...
-        case Expr.AppExpr(var pos, Expr.RawSortExpr(var uPos, var kind), var arg) when kind == SortKind.Type -> {
+        case Expr.AppExpr(var pos, Expr.RawSortExpr(var uPos, var kind), var arg)when kind == SortKind.Type -> {
           try {
             yield new Expr.TypeExpr(uPos, levelVar(arg.expr()));
           } catch (DesugarInterruption e) {
             yield new Expr.ErrorExpr(pos, expr);
           }
         }
-        case Expr.AppExpr(var pos, Expr.RawSortExpr(var uPos, var kind), var arg) when kind == SortKind.Set -> {
+        case Expr.AppExpr(var pos, Expr.RawSortExpr(var uPos, var kind), var arg)when kind == SortKind.Set -> {
```
Comment (ice1000) [inline base/src/main/java/org/aya/concrete/desugar/Desugarer.java:44]: 😅

File: parser/src/main/grammar/AyaPsiParser.bnf [hunk 2]
```diff
@@ -276,15 +286,15 @@ lambdaExpr ::= KW_LAMBDA lambdaTele+ (IMPLIES expr)?
 matchExpr ::= KW_MATCH exprList clauses
 doExpr ::= KW_DO LBRACE? doBlock RBRACE?
 thisExpr ::= KW_THIS (AT qualifiedId)?
-idiomExpr ::= LIDIOM idiomBlock? RIDIOM { pin=1 }
+idiomAtom ::= LIDIOM idiomBlock RIDIOM { pin=1 }
 pathExpr ::= LPATH pathTele+ RPATH expr partialExpr?
```
Comment (imkiva) [inline parser/src/main/grammar/AyaPsiParser.bnf:290]: I guess idioms can be empty as long as there exists an `empty` of that `Alternative`

Comment (imkiva) [reply parser/src/main/grammar/AyaPsiParser.bnf:290]: I will do it in a new PR

**Pull request merged**